### PR TITLE
convert CreditTransfer reducer to typescript and add a couple tests

### PR DESCRIPTION
**Pull request by gtaschuk**
**Describe the Pull Request**
converts the `creditTransfer` reducer to typescript following the format of the `messages` reducer, and adds a couple specs

**Q:**
~~I saw references to `PUSHER_*` actions that look unused ... are those deprecated now?~~

### CODE REVIEW

File: app/client/sagas/creditTransferSagas.js [hunk 1]
```diff
@@ -2,18 +2,27 @@ import { put, takeEvery, call, all } from "redux-saga/effects";
...
-  LOAD_CREDIT_TRANSFER_LIST_SUCCESS,
-  LOAD_CREDIT_TRANSFER_LIST_FAILURE,
-  MODIFY_TRANSFER_FAILURE,
-  MODIFY_TRANSFER_REQUEST,
-  MODIFY_TRANSFER_SUCCESS,
-  CREATE_TRANSFER_REQUEST,
-  CREATE_TRANSFER_SUCCESS,
-  CREATE_TRANSFER_FAILURE
-} from "../reducers/creditTransferReducer.js";
+  PUSHER_CREDIT_TRANSFER // TODO UNUSED?
```
Comment (tristanhcole) [inline app/client/sagas/creditTransferSagas.js:5]: `PUSHER_CREDIT_TRANSFER` is used to handle credit transfers received from the Pusher WebSocket on the dashboard page.

https://github.com/teamsempo/SempoBlockchain/blob/7588356216d645fbf3a998821d6f70215b0cccb7/app/client/components/pages/dashboardPage.jsx#L6

https://github.com/teamsempo/SempoBlockchain/blob/7588356216d645fbf3a998821d6f70215b0cccb7/app/client/components/pages/dashboardPage.jsx#L91-L116

Comment (tristanhcole) [reply app/client/sagas/creditTransferSagas.js:5]: `PUSHER_CREDIT_TRANSFER` cannot be resolved here so is causing the following error when testing frontend:

```
Error: takeEvery$1 requires a pattern or channel
    at check (io-6de156f3.js?86aa:33)
    at validateTakeEffect (redux-saga-effects.esm.js?fc1c:382)
    at takeEvery$1 (redux-saga-effects.esm.js?fc1c:388)
    at watchPusherCreditTransfer$ (creditTransferSagas.js?baa4:135)
    at tryCatch (runtime.js?f3a2:65)
    at Generator.invoke [as _invoke] (runtime.js?f3a2:303)
    at Generator.prototype.<computed> [as next] (runtime.js?f3a2:117)
    at next (redux-saga-core.esm.js?42a9:1157)
    at proc (redux-saga-core.esm.js?42a9:1108)
    at runEffect (redux-saga-core.esm.js?42a9:1199)
```

Comment (gtaschuk) [reply app/client/sagas/creditTransferSagas.js]: Oops - fixed now.  Not sure how I missed that.

So how are events being pushed into this websocket subscription?

File: app/client/reducers/creditTransfer/actions.ts [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+  credit_transfers: CreditTransfers
+) => ({
+  type: LoadCreditTransferActionTypes.LOAD_CREDIT_TRANSFER_LIST_SUCCESS as typeof LoadCreditTransferActionTypes.LOAD_CREDIT_TRANSFER_LIST_SUCCESS,
+  credit_transfers
+});
+
+export const loadCreditTransferListFailure = (err: any) => ({
+  type: LoadCreditTransferActionTypes.LOAD_CREDIT_TRANSFER_LIST_FAILURE as typeof LoadCreditTransferActionTypes.LOAD_CREDIT_TRANSFER_LIST_FAILURE,
+  error: err
+});
```
[the remark this answers is not in the document]
Comment (gtaschuk) [reply app/client/reducers/creditTransfer/actions.ts:25]: Makes sense.  I changed the way they are referenced from the saga (see commit 3) to match

It's a bit wordy, maybe worth renaming `CreditTransferAction.updateCreditTransferListRequest(credit_transfers)` to `CreditTransferActon.updateListRequest(credit_transfers)` in the future?

[the remark this answers is not in the document]
Comment (tristanhcole) [reply app/client/reducers/creditTransfer/actions.ts:25]: > It's a bit wordy, maybe worth renaming CreditTransferAction.updateCreditTransferListRequest(credit_transfers) to CreditTransferActon.updateListRequest(credit_transfers) in the future?

Good point. Would be even cleaner

File: app/client/sagas/creditTransferSagas.js (hunk 1 above)
Comment (tristanhcole) [reply app/client/sagas/creditTransferSagas.js]: They're pushed to the frontend after a successful request:
https://github.com/teamsempo/SempoBlockchain/blob/7588356216d645fbf3a998821d6f70215b0cccb7/app/server/__init__.py#L115-L132

**Comment by gtaschuk** [comment]
@tristanhcole Fixed pusher events and converted the relevant saga to typescript too

File: app/client/sagas/transferAccountSagas.ts [hunk 3]
```diff
@@ -57,8 +70,13 @@ function* updateStateFromTransferAccount(data) {
   }
 }
 
+interface TransferAccountLoadApiResult {
+  type: typeof LOAD_TRANSFER_ACCOUNTS_REQUEST;
+  payload: any;
```
Comment (tristanhcole) [inline app/client/sagas/transferAccountSagas.ts:75]: `payload` of `any` kind of defeats the purpose of typescript imo.

File: app/client/reducers/creditTransfer/actions.ts [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+  CreditTransfers,
+  CreditTransferActionTypes,
+  LoadCreditTransferActionTypes,
+  ModifyCreditTransferActionTypes
+} from "./types";
+
+// ACTIONS
+export const loadCreditTransferListRequest = () => ({
+  type: LoadCreditTransferActionTypes.LOAD_CREDIT_TRANSFER_LIST_REQUEST as typeof LoadCreditTransferActionTypes.LOAD_CREDIT_TRANSFER_LIST_REQUEST
+});
```
Comment (tristanhcole) [inline app/client/reducers/creditTransfer/actions.ts:13]: This needs a payload. Incorrectly refactored on dashboard page also.

File: app/client/reducers/creditTransfer/actions.ts [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+
+export type ModifyCreditTransferAction = ActionsUnion<
+  typeof ModifyCreditTransferAction
+>;
+
+export const createTransferRequest = (payload: any) => ({
+  type: CreditTransferActionTypes.CREATE_TRANSFER_REQUEST as typeof CreditTransferActionTypes.CREATE_TRANSFER_REQUEST,
+  payload
+});
+
```
Comment (tristanhcole) [inline app/client/reducers/creditTransfer/actions.ts:62]: `payload` of `any` kind of defeats the purpose of typescript imo.

File: app/client/reducers/creditTransfer/actions.ts [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+export const createTransferSuccess = () => ({
+  type: CreditTransferActionTypes.CREATE_TRANSFER_SUCCESS as typeof CreditTransferActionTypes.CREATE_TRANSFER_SUCCESS
+});
+
+export const createTransferFailure = (error: any) => ({
+  type: CreditTransferActionTypes.CREATE_TRANSFER_FAILURE as typeof CreditTransferActionTypes.CREATE_TRANSFER_FAILURE,
+  error
+});
```
Comment (tristanhcole) [inline app/client/reducers/creditTransfer/actions.ts:70]: should be string

**Comment by tristanhcole** [comment]
> @tristanhcole Fixed pusher events and converted the relevant saga to typescript too

yep, reviewing and a few small fixes

File: app/client/components/creditTransfer/creditTransferList.jsx [hunk 7]
```diff
@@ -5,7 +5,7 @@ import ReactTable from "react-table";
 
 import { TopRow, StyledSelect } from "../styledElements.js";
 
-import { modifyTransferRequest } from "../../reducers/creditTransferReducer";
+import { ModifyCreditTransferAction } from "../../reducers/creditTransfer/actions";
```
Comment (tristanhcole) [inline app/client/components/creditTransfer/creditTransferList.jsx:8]: This was causing an error

File: app/client/reducers/creditTransfer/actions.ts [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+  loadCreditTransferListSuccess: () =>
+    createAction(
+      LoadCreditTransferActionTypes.LOAD_CREDIT_TRANSFER_LIST_SUCCESS
+    ),
+  loadCreditTransferListFailure: (err: string) =>
+    createAction(
+      LoadCreditTransferActionTypes.LOAD_CREDIT_TRANSFER_LIST_FAILURE,
+      err
+    )
+};
```
Comment (tristanhcole) [inline app/client/reducers/creditTransfer/actions.ts:28]: @gtaschuk I refactored to using the `createAction` generic. Can you please do the same in https://github.com/teamsempo/SempoBlockchain/pull/298 for readability/maintainability

File: app/client/components/transferAccount/transferAccountManager.jsx [hunk 9]
```diff
@@ -159,29 +156,6 @@ class TransferAccountManager extends React.Component {
     this.handleClick();
   }
 
-  createNewTransfer() {
```
Comment (tristanhcole) [inline app/client/components/transferAccount/transferAccountManager.jsx:162]: Unused code

File: app/client/sagas/transferAccountSagas.ts [hunk 10]
```diff
@@ -89,8 +101,13 @@ function* watchLoadTransferAccounts() {
   yield takeEvery(LOAD_TRANSFER_ACCOUNTS_REQUEST, loadTransferAccounts);
 }
 
+interface TransferAccountEditApiResult {
+  type: typeof EDIT_TRANSFER_ACCOUNT_REQUEST;
+  payload: any;
+}
+
```
Comment (tristanhcole) [inline app/client/sagas/transferAccountSagas.ts:108]: lets move this to the TransferAccount types in https://github.com/teamsempo/SempoBlockchain/pull/298
